### ISSUE TITLE
`.env`: Make MAILER_URL and DATABASE_URL consistent

### ISSUE THREAD

**Issue by ThomasLandauer** [symfony/symfony#30316]
**Description**  
Use either the form `username:password@server` **or** `server?username=&password=` for all connection url strings in `.env`

https://symfony.com/doc/current/doctrine.html#configuring-the-database says

> DATABASE_URL="mysql://db_user:db_password@127.0.0.1:3306/db_name"

whereas https://symfony.com/doc/current/email.html#configuration says

> MAILER_URL=smtp://localhost:25?encryption=ssl&auth_mode=login&username=&password=

And (weirdly enough), the `.env.dist` file itself contains this example:

> For Gmail as a transport, use: "gmail://username:password@localhost"

So I'm suggesting to make it consistent for all!

Besides, how to escape special characters should be explained better. https://symfony.com/doc/current/doctrine.html#configuring-the-database has a dedicated alert box about this, https://symfony.com/doc/current/email.html#configuration just has a comment in the code. So I'd suggest to copy the alert box to the email docs as well. Should I do that?

Is the parsing of the `.env` file done per package or centrally? If centrally, then maybe the escaping question should rather be explained at https://symfony.com/doc/current/components/dotenv.html ?

**Comment by xabbuh** [comment]
Well, the issue is that these URLs are interpreted by different systems which follow different formats. To be able to have a universal format would mean that we would have to somewhere transform them which will probably sooner or later lead to confusion when the format used here does not match the format used in the documentation of the used libraries.

**Comment by ThomasLandauer** [comment]
OK, thanks, so it's not done centrally ;-)

What about opening an issue at Doctrine bundle or Swiftmailer bundle to change the format? Which format would you say is better: `username:password@server` or `server?username=&password=`?

Any why didn't the guys who invented this, make it consistent in the first place? ;-)

**Comment by nicolas-grekas** [comment]
> Any why didn't the guys who invented this, make it consistent in the first place? ;-)

because separate individuals don't share the same brain ;) If you think this should be done, I think you should send a PR. That's the best way to prove your point, as always in OSS ;)

**Comment by ThomasLandauer** [comment]
I updated the Swift Mailer docs about the special characters: https://github.com/symfony/symfony-docs/pull/11054

And I suggested to change the syntax in Swift Mailer Bundle: https://github.com/symfony/swiftmailer-bundle/issues/276  
Main reasons for suggesting Swift Mailer to adopt Doctrine's syntax (and not vice versa):
* Swift Mailer is inconsistent in itself, since it's already using the Doctrine syntax for `gmail://`.
* The Doctrine syntax has (some) RFC support.

So I'm closing this.
